[docs] Improve the docs for the filtering feature (#945)
mui-org/material-ui-x: docs/src/pages/components/data-grid/filtering/filtering.md

The data grid supports different operators for the native column types.
However, you can extend the operator and add your own, customize the input component or set your own operator for a new column type.

1. **Custom input**.

> In this demo, you will see how to reuse the numeric filter and customize the input filter value component.

Reviewer: ```suggestion
In this demo you will see how to reuse the numeric filter and customize the input filter value component.
```

The rating column reuses the numeric operator, but the input value is a new rating component.

{{"demo": "pages/components/data-grid/filtering/ExtendNumericOperator.js", "bg": "inline"}}

2. **Custom column type**.

> In this demo, you will see how to extend an existing column type, add your own filter operators with filter input value props.

Reviewer: ```suggestion
In this demo you will see how to extend an existing column type, adding your own filter operators with filter input value props.
```

> As you can see in the filter panel, the `totalPrice` column only contains 2 operators `<`, `>`, and the input field is prefixed with `$`.

Reviewer: ```suggestion
As you can see in the filter panel, the `totalPrice` column only contains 2 operators `<`, & `>`, and the input field is prefixed with `$`.
```

{{"demo": "pages/components/data-grid/filtering/ColumnTypeFilteringGrid.js", "bg": "inline"}}

3. **Custom operator**.

> In this demo, you will see how to create a complete new operator for a specific column.

Reviewer: ```suggestion
In this demo you will see how to create a complete new operator for a specific column.
```

> The rating column contains new `From` operator as you can see in the filter panel.

Reviewer: ```suggestion
The rating column contains a new `From` operator, as you can see in the filter panel.
```

{{"demo": "pages/components/data-grid/filtering/CustomRatingOperator.js", "bg": "inline"}}

### Server-side filter

> As in the sort feature, filter can be run server side by setting the `filterMode` prop to `server`, and implementing the `onFilterChange` handler.

Reviewer: ```suggestion
As with the sort feature, filtering can be run server side by setting the `filterMode` prop to `server`, and implementing the `onFilterChange` handler.
```

> As with the sort feature, filtering can be run server side by setting the `filterMode` prop to `server`, and implementing the `onFilterChange` handler.

Reviewer 2: I don't see why we should mention sorting over pagination or else. I have also fixed the name of the prop, it's `onFilterModelChange`.

```suggestion
Filtering can be run server-side by setting the `filterMode` prop to `server`, and implementing the `onFilterModelChange` handler.
```

```tsx
<DataGrid
  rows={rows}
  columns={columns}
  filterMode="server"

>   onFilterModelChange={onFilterChange}

Reviewer 2: I think that renaming variable like this creates confusion

```suggestion
  onFilterModelChange={onFilterModelChange}
```

Reviewer 2: or if we want to follow the popular handle convention:
```suggestion
  onFilterModelChange={handleFilterModelChange}
```
